tests/baselines/reference/declarationEmitTypeParameterNameShadowedInternally.types

```diff
@@ -0,0 +1,21 @@
+//// [tests/cases/compiler/declarationEmitTypeParameterNameShadowedInternally.ts] ////
+
+=== declarationEmitTypeParameterNameShadowedInternally.ts ===
+export const foo = <T,>(x: T) => {
+>foo : <T>(x: T) => <T>(y: T) => readonly [T, T]
+><T,>(x: T) => {	const inner = <T,>(y: T) => [x, y] as const;	return inner;} : <T>(x: T) => <T>(y: T) => readonly [T, T]
+>x : T
+
+	const inner = <T,>(y: T) => [x, y] as const;
+>inner : <T>(y: T) => readonly [T, T]
+><T,>(y: T) => [x, y] as const : <T>(y: T) => readonly [T, T]
+>y : T
+>[x, y] as const : readonly [T, T]
+>[x, y] : readonly [T, T]
+>x : T
+>y : T
+
+	return inner;
+>inner : <T>(y: T) => readonly [T, T]

```

Comment:
Here you can see that the tooltip is very wrong because we don't do the right renaming there...

Reply:
Nothing new, though:
![image](https://github.com/microsoft/TypeScript/assets/5341706/2279055d-eb96-42e3-b198-e701ff88d380)